### PR TITLE
Alerting: Add debug logs for EndsAt timestamp

### PR DESCRIPTION
**Pull request by grobinson-grafana**
**What is this feature?**

This commit adds debug logs for previous_ends_at and next_ends_at to state.go to help us debug issues where alerts are resolved in Alertmanager due to expiration. This change is in response to a support escalation where this information was needed but unavailable.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ngalert/state/state.go [hunk 1]
```diff
@@ -138,8 +138,10 @@ func (a *State) Resolve(reason string, endsAt time.Time) {
 }
 
 // Maintain updates the end time using the most recent evaluation.
-func (a *State) Maintain(interval int64, evaluatedAt time.Time) {
+func (a *State) Maintain(interval int64, evaluatedAt time.Time) time.Time {
```
Comment (JacobsonMT) [inline pkg/services/ngalert/state/state.go:141]: NIT: I don't think we need to change the signature of `Maintain` to return `EndsAt` considering we have access to that via `state.EndsAt` at the point in the code that we're doing the logging. I don't mind either way though, if you have a stronger opinion here, that's fine with me and keep as is.

**Review by JacobsonMT — APPROVED** [review]
LGTM with one optional NIT :rocket:

File: pkg/services/ngalert/state/state.go (hunk 1 above)
Comment (grobinson-grafana) [reply pkg/services/ngalert/state/state.go:141]: Ah, so the reason I did this is so we wouldn't need to do `oldEndsAt := state.EndsAt` each time. Do you think that's better?

**Comment by grobinson-grafana** [comment]
https://github.com/grafana/support-escalations/issues/6191

**Pull request merged**